pkg/ccrm/ccrm_cloudletapi.go

```diff
@@ -144,19 +142,32 @@ func (s *CCRMHandler) NameSanitize(ctx context.Context, in *edgeproto.NameSaniti
 	}, nil
 }
 
-func (s *CCRMHandler) getCloudletPlatform(ctx context.Context, key *edgeproto.CloudletKey) (*edgeproto.Cloudlet, platform.Platform, error) {
-	if key == nil {
-		return nil, nil, fmt.Errorf("CloudletKey not specified")
+func (s *CCRMHandler) RefreshCerts(ctx context.Context, in *edgeproto.Cloudlet) (*edgeproto.Result, error) {
+	pf, err := s.getCRMCloudletPlatform(ctx, &in.Key)
+	if err != nil {
+		return nil, err
+	}
+	features := pf.GetFeatures()
+
+	proxyCerts := certs.NewProxyCerts(ctx, &in.Key, pf, s.nodeMgr, nil, features, s.flags.CommercialCerts, s.flags.EnvoyWithCurlImage, s.proxyCertsCache)
```

Comment:
Why not keep the proxyCerts as part of the CCRMHandler and just  start the refresh thread here instead of delegating this to ccrm_periodic?

Reply:
For load balancing. ccrm_periodic distributes the refresh for each cloudlet to different CCRMs. If we just did it here, this CCRM would have to run the refresh for every cloudlet.

Reply:
ah, ok. Otherwise all ccrm instances will run this thread. Makes sense.